### ISSUE TITLE
Add childResource to OAM definition spec

### ISSUE THREAD

**Issue by resouer** [oam-dev/spec#410]
It has been proved that childResource is a strong need for any abstraction workload (e.g. ContainerizedWorkload), so that Trait controller could know what's the underlying workload object it need to handle with.

Note that this field is Kubernetes specific for now, but as a runtime configuration object, it's reasonable that definition object has runtime specific field.

**Comment by ryanzhang-oss** [comment]
I think it's better to create a separate issue on the "runtime", it deserves its own issue :)

**Comment by wonderflow** [comment]
I agree this proposal, what about make `runtimePlatform: Kubernetes` as default?
